service/src/main/java/bio/terra/cbas/dao/util/SqlPlaceholderMapping.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.cbas.dao.util;
+
+import bio.terra.cbas.util.Pair;
+import com.google.common.collect.Streams;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class SqlPlaceholderMapping<A> {
```

Comment:
curious to learn more about this -- is this your solution for "a neater way to build SQL queries"?

Reply:
It's a small step in that direction, yeah. Still a little more bespoke than I'd like but it seems to be working for now, and definitely simplified the in-DAO logic, which was nice